pkg/agent/types/knapsack.go

```diff
@@ -0,0 +1,30 @@
+package types
+
+import "go.etcd.io/bbolt"
+
+// Knapsack is an inventory of data and useful services which are used throughout
+// launcher code and are typically valid for the lifetime of the launcher application instance.
+type Knapsack struct {
```

Comment:
I think there's a sprawling conversation around struct vs type here, but I'm happy to let that keep evolving.